ipaserver/plugins/cert.py

```diff
@@ -953,6 +955,27 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
         )
 
 
+def _emails_are_valid(csr_emails, principal_emails):
+    """
+    Checks if any email address from certificate does not
+    appear in ldap entry, comparing the domain part case-insensitively.
+    """
+
+    if not any(principal_emails) or not any(csr_emails):
+        return False
```

Comment:
We should get rid of this.  The subject check is sufficient.  If there are no `csr_emails`, then
the test should be vacuously `True`, not `False`.  Similarly, if there are no `principal_emails`, then
the valid if and only if there are also no `csr_email`.  All of this is covered by the subset check,
so just remove these two lines.